File: beams/beams/custom_scripts/training_program/training_program.py
import frappe
import re

def validate_training_program(doc, method):
    """ Validation for Email and Contact Number"""
    
    if doc.trainer_email and not frappe.utils.validate_email_address(doc.trainer_email):
        frappe.throw("Please enter a valid email address.")

    if doc.contact_number and not re.fullmatch(r"^\d{10}$", doc.contact_number):
        frappe.throw("Please enter a valid 10-digit contact number.")

File: beams/beams/custom_scripts/training_program/training_program.js

frappe.ui.form.on('Training Program', {
    validate: function(frm) {
        // Validate Email Format
        if (frm.doc.trainer_email && !frappe.utils.validate_type(frm.doc.trainer_email, "email")) {
            frappe.msgprint(__('Please enter a valid email address'));
            frappe.validated = false;
        }

        // Validate Contact Number (Only 10-digit numbers)
        if (frm.doc.contact_number && !/^\d{10}$/.test(frm.doc.contact_number)) {
            frappe.msgprint(__('Please enter a valid 10-digit contact number'));
            frappe.validated = false;
        }
    }
});
